*** ./smali/com/android/internal/widget/ILockSettings$Stub.smali	2015-08-27 23:29:11.775027681 +0800
--- /home/nian/patchrom/m7/temp/new_smali/framework2.jar.out/./smali/com/android/internal/widget/ILockSettings$Stub.smali	2015-08-27 23:30:29.827026641 +0800
***************
*** 52,57 ****
--- 52,59 ----
  
  .field static final TRANSACTION_setLong:I = 0x2
  
+ .field static final TRANSACTION_setRawLockPassword:I = 0xf
+ 
  .field static final TRANSACTION_setString:I = 0x3
  
  
***************
*** 578,583 ****
--- 580,606 ----
  
      nop
  
+     :sswitch_miui_0
+     const-string v7, "com.android.internal.widget.ILockSettings"
+ 
+     invoke-virtual {p2, v7}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
+ 
+     invoke-virtual {p2}, Landroid/os/Parcel;->createByteArray()[B
+ 
+     move-result-object v0
+ 
+     .local v0, "_arg0":[B
+     invoke-virtual {p2}, Landroid/os/Parcel;->readInt()I
+ 
+     move-result v1
+ 
+     .restart local v1    # "_arg1":I
+     invoke-virtual {p0, v0, v1}, Lcom/android/internal/widget/ILockSettings$Stub;->setRawLockPassword([BI)V
+ 
+     invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
+ 
+     goto/16 :goto_0
+ 
      :sswitch_data_0
      .sparse-switch
          0x1 -> :sswitch_1
***************
*** 594,599 ****
--- 617,623 ----
          0xc -> :sswitch_c
          0xd -> :sswitch_d
          0xe -> :sswitch_e
+         0xf -> :sswitch_miui_0
          0x5f4e5446 -> :sswitch_0
      .end sparse-switch
  .end method
