*** ./smali/com/android/internal/widget/LockPatternView.smali	2015-08-27 23:29:11.415027685 +0800
--- /home/nian/patchrom/m7/temp/new_smali/framework2.jar.out/./smali/com/android/internal/widget/LockPatternView.smali	2015-08-27 23:30:29.051026652 +0800
***************
*** 3403,3409 ****
  .end method
  
  .method protected onRestoreInstanceState(Landroid/os/Parcelable;)V
!     .locals 5
      .param p1, "state"    # Landroid/os/Parcelable;
  
      .prologue
--- 3403,3409 ----
  .end method
  
  .method protected onRestoreInstanceState(Landroid/os/Parcelable;)V
!     .locals 3
      .param p1, "state"    # Landroid/os/Parcelable;
  
      .prologue
***************
*** 3434,3444 ****
  
      invoke-virtual {v0}, Lcom/android/internal/widget/LockPatternView$SavedState;->getSerializedPattern()Ljava/lang/String;
  
!     move-result-object v3
! 
!     iget-byte v4, p0, Lcom/android/internal/widget/LockPatternView;->mPatternSize:B
  
!     invoke-virtual {v2, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->stringToPattern(Ljava/lang/String;B)Ljava/util/List;
  
      move-result-object v2
  
--- 3434,3442 ----
  
      invoke-virtual {v0}, Lcom/android/internal/widget/LockPatternView$SavedState;->getSerializedPattern()Ljava/lang/String;
  
!     move-result-object v2
  
!     invoke-static {v2}, Lcom/android/internal/widget/LockPatternUtils;->stringToPattern(Ljava/lang/String;)Ljava/util/List;
  
      move-result-object v2
  
***************
*** 3535,3545 ****
      :cond_0
      iget-object v2, p0, Lcom/android/internal/widget/LockPatternView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
!     iget-object v3, p0, Lcom/android/internal/widget/LockPatternView;->mPattern:Ljava/util/ArrayList;
! 
!     iget-byte v4, p0, Lcom/android/internal/widget/LockPatternView;->mPatternSize:B
  
!     invoke-virtual {v2, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->patternToString(Ljava/util/List;B)Ljava/lang/String;
  
      move-result-object v2
  
--- 3533,3541 ----
      :cond_0
      iget-object v2, p0, Lcom/android/internal/widget/LockPatternView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
!     iget-object v2, p0, Lcom/android/internal/widget/LockPatternView;->mPattern:Ljava/util/ArrayList;
  
!     invoke-static {v2}, Lcom/android/internal/widget/LockPatternUtils;->patternToString(Ljava/util/List;)Ljava/lang/String;
  
      move-result-object v2
  
