*** ./smali/android/preference/RingtonePreference.smali	2015-08-27 23:29:02.375027806 +0800
--- /home/nian/patchrom/m7/temp/new_smali/framework.jar.out/./smali/android/preference/RingtonePreference.smali	2015-08-27 23:30:16.971026813 +0800
***************
*** 363,368 ****
--- 363,374 ----
  
      invoke-virtual {p1, v0, v1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/lang/CharSequence;)Landroid/content/Intent;
  
+     const-string v0, "com.android.thememanager"
+ 
+     const-string v1, "com.android.thememanager.activity.ThemeTabActivity"
+ 
+     invoke-virtual {p1, v0, v1}, Landroid/content/Intent;->setClassName(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
+ 
      return-void
  
      :cond_2
