*** ./smali/android/view/ViewRootImpl.smali	2015-08-27 23:28:25.111028302 +0800
--- /home/nian/patchrom/m7/temp/new_smali/framework.jar.out/./smali/android/view/ViewRootImpl.smali	2015-08-27 23:29:45.827027227 +0800
***************
*** 9142,9147 ****
--- 9142,9165 ----
      return-void
  .end method
  
+ .method createInputChannelAnyWay()V
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Landroid/view/ViewRootImpl;->mInputChannel:Landroid/view/InputChannel;
+ 
+     if-nez v0, :cond_0
+ 
+     new-instance v0, Landroid/view/InputChannel;
+ 
+     invoke-direct {v0}, Landroid/view/InputChannel;-><init>()V
+ 
+     iput-object v0, p0, Landroid/view/ViewRootImpl;->mInputChannel:Landroid/view/InputChannel;
+ 
+     :cond_0
+     return-void
+ .end method
+ 
  .method public debug()V
      .locals 1
  
***************
*** 9350,9355 ****
--- 9368,9393 ----
      goto :goto_1
  .end method
  
+ .method discardInputChannelBySetting()V
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Landroid/view/ViewRootImpl;->mWindowAttributes:Landroid/view/WindowManager$LayoutParams;
+ 
+     iget v0, v0, Landroid/view/WindowManager$LayoutParams;->inputFeatures:I
+ 
+     and-int/lit8 v0, v0, 0x2
+ 
+     if-eqz v0, :cond_0
+ 
+     const/4 v0, 0x0
+ 
+     iput-object v0, p0, Landroid/view/ViewRootImpl;->mInputChannel:Landroid/view/InputChannel;
+ 
+     :cond_0
+     return-void
+ .end method
+ 
  .method public dispatchAppVisibility(Z)V
      .locals 3
      .param p1, "visible"    # Z
***************
*** 13040,13047 ****
  
      if-eq v0, p1, :cond_0
  
-     iput-boolean p1, p0, Landroid/view/ViewRootImpl;->mStopped:Z
- 
      if-nez p1, :cond_0
  
      invoke-virtual {p0}, Landroid/view/ViewRootImpl;->scheduleTraversals()V
--- 13078,13083 ----
***************
*** 13377,13386 ****
      move-object/from16 v0, p0
  
      iput-object v3, v0, Landroid/view/ViewRootImpl;->mInputChannel:Landroid/view/InputChannel;
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
-     :cond_6
      :try_start_1
      move-object/from16 v0, p0
  
--- 13413,13424 ----
      move-object/from16 v0, p0
  
      iput-object v3, v0, Landroid/view/ViewRootImpl;->mInputChannel:Landroid/view/InputChannel;
+ 
+     :cond_6
+     invoke-virtual/range {p0 .. p0}, Landroid/view/ViewRootImpl;->createInputChannelAnyWay()V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
      :try_start_1
      move-object/from16 v0, p0
  
***************
*** 13957,13962 ****
--- 13995,14002 ----
      iput-object v3, v0, Landroid/view/ViewRootImpl;->mInputQueueCallback:Landroid/view/InputQueue$Callback;
  
      :cond_d
+     invoke-virtual/range {p0 .. p0}, Landroid/view/ViewRootImpl;->discardInputChannelBySetting()V
+ 
      move-object/from16 v0, p0
  
      iget-object v3, v0, Landroid/view/ViewRootImpl;->mInputChannel:Landroid/view/InputChannel;
