*** ./smali/com/android/server/AlarmManagerService.smali	2015-08-27 23:27:43.947028850 +0800
--- /home/nian/patchrom/m7/temp/new_smali/services.jar.out/./smali/com/android/server/AlarmManagerService.smali	2015-08-27 23:29:15.299027634 +0800
***************
*** 7,12 ****
--- 7,13 ----
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
          Lcom/android/server/AlarmManagerService$ResultReceiver;,
+         Lcom/android/server/AlarmManagerService$Injector;,
          Lcom/android/server/AlarmManagerService$UninstallReceiver;,
          Lcom/android/server/AlarmManagerService$ClockReceiver;,
          Lcom/android/server/AlarmManagerService$QuickBootReceiver;,
***************
*** 84,91 ****
      .end annotation
  .end field
  
- .field private final mAppOps:Landroid/app/AppOpsManager;
- 
  .field private final mBlockedUids:Ljava/util/ArrayList;
      .annotation system Ldalvik/annotation/Signature;
          value = {
--- 85,90 ----
***************
*** 417,434 ****
      invoke-virtual {v3}, Lcom/android/server/AlarmManagerService$AlarmThread;->start()V
  
      :goto_0
-     iget-object v3, p0, Lcom/android/server/AlarmManagerService;->mContext:Landroid/content/Context;
- 
-     const-string v4, "appops"
- 
-     invoke-virtual {v3, v4}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
- 
-     move-result-object v3
- 
-     check-cast v3, Landroid/app/AppOpsManager;
- 
-     iput-object v3, p0, Lcom/android/server/AlarmManagerService;->mAppOps:Landroid/app/AppOpsManager;
- 
      return-void
  
      :cond_1
--- 416,421 ----
***************
*** 582,597 ****
      return-object v0
  .end method
  
- .method static synthetic access$1800(Lcom/android/server/AlarmManagerService;)Landroid/app/AppOpsManager;
-     .locals 1
-     .param p0, "x0"    # Lcom/android/server/AlarmManagerService;
- 
-     .prologue
-     iget-object v0, p0, Lcom/android/server/AlarmManagerService;->mAppOps:Landroid/app/AppOpsManager;
- 
-     return-object v0
- .end method
- 
  .method static synthetic access$1900(Lcom/android/server/AlarmManagerService;)Ljava/util/HashMap;
      .locals 1
      .param p0, "x0"    # Lcom/android/server/AlarmManagerService;
--- 569,574 ----
***************
*** 1449,1455 ****
      goto :goto_0
  .end method
  
! .method private removeWithReportLocked(Landroid/app/PendingIntent;)Z
      .locals 16
      .param p1, "operation"    # Landroid/app/PendingIntent;
  
--- 1426,1432 ----
      goto :goto_0
  .end method
  
! .method public removeLocked(Landroid/app/PendingIntent;)V
      .locals 16
      .param p1, "operation"    # Landroid/app/PendingIntent;
  
***************
*** 1643,1649 ****
      invoke-direct/range {p0 .. p0}, Lcom/android/server/AlarmManagerService;->rescheduleKernelAlarmsLocked()V
  
      :cond_4
!     return v11
  .end method
  
  .method private rescheduleKernelAlarmsLocked()V
--- 1620,1626 ----
      invoke-direct/range {p0 .. p0}, Lcom/android/server/AlarmManagerService;->rescheduleKernelAlarmsLocked()V
  
      :cond_4
!     return-void
  .end method
  
  .method private rescheduleKernelAlarmsLocked()V
***************
*** 1763,1769 ****
  .method private native set(IIJJ)V
  .end method
  
! .method private setImplLocked(IJJJJJLandroid/app/PendingIntent;ZZLandroid/os/WorkSource;Z)V
      .locals 22
      .param p1, "type"    # I
      .param p2, "when"    # J
--- 1740,1746 ----
  .method private native set(IIJJ)V
  .end method
  
! .method private setImplLocked(IJJJJJLandroid/app/PendingIntent;ZZLandroid/os/WorkSource;)V
      .locals 22
      .param p1, "type"    # I
      .param p2, "when"    # J
***************
*** 1775,1781 ****
      .param p13, "isStandalone"    # Z
      .param p14, "doValidate"    # Z
      .param p15, "workSource"    # Landroid/os/WorkSource;
-     .param p16, "wakeupFiltered"    # Z
  
      .prologue
      new-instance v5, Lcom/android/server/AlarmManagerService$Alarm;
--- 1752,1757 ----
***************
*** 1803,1834 ****
  
      move-object/from16 v1, p12
  
!     invoke-direct {v0, v1}, Lcom/android/server/AlarmManagerService;->removeWithReportLocked(Landroid/app/PendingIntent;)Z
! 
!     move-result v20
! 
!     .local v20, "foundExistingWakeup":Z
!     if-nez v20, :cond_0
! 
!     if-eqz p16, :cond_0
! 
!     move-object/from16 v0, p0
  
-     iget-object v6, v0, Lcom/android/server/AlarmManagerService;->mAppOps:Landroid/app/AppOpsManager;
- 
-     const/16 v7, 0x2e
- 
-     invoke-virtual/range {p12 .. p12}, Landroid/app/PendingIntent;->getCreatorUid()I
- 
-     move-result v8
- 
-     invoke-virtual/range {p12 .. p12}, Landroid/app/PendingIntent;->getCreatorPackage()Ljava/lang/String;
- 
-     move-result-object v9
- 
-     invoke-virtual {v6, v7, v8, v9}, Landroid/app/AppOpsManager;->noteOpNoThrow(IILjava/lang/String;)I
- 
-     :cond_0
      if-eqz p13, :cond_2
  
      const/16 v21, -0x1
--- 1779,1786 ----
  
      move-object/from16 v1, p12
  
!     invoke-virtual {v0, v1}, Lcom/android/server/AlarmManagerService;->removeLocked(Landroid/app/PendingIntent;)V
  
      if-eqz p13, :cond_2
  
      const/16 v21, -0x1
***************
*** 2204,2218 ****
  
      move-object/from16 v21, v0
  
-     const/16 v22, 0x0
- 
      move-object/from16 v6, p0
  
      move v7, v10
  
      move-wide v10, v4
  
!     invoke-direct/range {v6 .. v22}, Lcom/android/server/AlarmManagerService;->setImplLocked(IJJJJJLandroid/app/PendingIntent;ZZLandroid/os/WorkSource;Z)V
  
      .end local v4    # "nextElapsed":J
      .end local v26    # "delta":J
--- 2156,2168 ----
  
      move-object/from16 v21, v0
  
      move-object/from16 v6, p0
  
      move v7, v10
  
      move-wide v10, v4
  
!     invoke-direct/range {v6 .. v21}, Lcom/android/server/AlarmManagerService;->setImplLocked(IJJJJJLandroid/app/PendingIntent;ZZLandroid/os/WorkSource;)V
  
      .end local v4    # "nextElapsed":J
      .end local v26    # "delta":J
***************
*** 3544,3558 ****
  
      move-object/from16 v21, v0
  
-     const/16 v22, 0x0
- 
      move-object/from16 v6, p0
  
      move-wide v10, v4
  
      move/from16 v20, p1
  
!     invoke-direct/range {v6 .. v22}, Lcom/android/server/AlarmManagerService;->setImplLocked(IJJJJJLandroid/app/PendingIntent;ZZLandroid/os/WorkSource;Z)V
  
      add-int/lit8 v27, v27, 0x1
  
--- 3494,3506 ----
  
      move-object/from16 v21, v0
  
      move-object/from16 v6, p0
  
      move-wide v10, v4
  
      move/from16 v20, p1
  
!     invoke-direct/range {v6 .. v21}, Lcom/android/server/AlarmManagerService;->setImplLocked(IJJJJJLandroid/app/PendingIntent;ZZLandroid/os/WorkSource;)V
  
      add-int/lit8 v27, v27, 0x1
  
***************
*** 3747,3762 ****
      throw v0
  .end method
  
- .method public removeLocked(Landroid/app/PendingIntent;)V
-     .locals 0
-     .param p1, "operation"    # Landroid/app/PendingIntent;
- 
-     .prologue
-     invoke-direct {p0, p1}, Lcom/android/server/AlarmManagerService;->removeWithReportLocked(Landroid/app/PendingIntent;)Z
- 
-     return-void
- .end method
- 
  .method public removeLocked(Ljava/lang/String;)V
      .locals 4
      .param p1, "packageName"    # Ljava/lang/String;
--- 3695,3700 ----
***************
*** 3950,3955 ****
--- 3888,3915 ----
      return-void
  
      :cond_0
+     move-object/from16 v0, p0
+ 
+     iget-object v3, v0, Lcom/android/server/AlarmManagerService;->mContext:Landroid/content/Context;
+ 
+     move/from16 v4, p1
+ 
+     move-wide/from16 v5, p2
+ 
+     move-wide/from16 v7, p4
+ 
+     move-wide/from16 v9, p6
+ 
+     move-object/from16 v11, p8
+ 
+     move/from16 v12, p9
+ 
+     move-object/from16 v13, p10
+ 
+     invoke-static/range {v3 .. v13}, Lcom/android/server/ExtraAlarmManagerService;->recalculateWindowLength(Landroid/content/Context;IJJJLandroid/app/PendingIntent;ZLandroid/os/WorkSource;)J
+ 
+     move-result-wide p4
+ 
      const-wide/32 v7, 0x2932e00
  
      cmp-long v7, p4, v7
***************
*** 4126,4179 ****
  
      .local v15, "maxElapsed":J
      :goto_1
-     const/16 v23, 0x0
- 
-     .local v23, "wakeupFiltered":Z
-     invoke-virtual/range {p8 .. p8}, Landroid/app/PendingIntent;->getCreatorUid()I
- 
-     move-result v7
- 
-     const/16 v8, 0x2710
- 
-     if-lt v7, v8, :cond_6
- 
-     if-eqz p1, :cond_5
- 
-     const/4 v7, 0x2
- 
-     move/from16 v0, p1
- 
-     if-ne v0, v7, :cond_6
- 
-     :cond_5
-     move-object/from16 v0, p0
- 
-     iget-object v7, v0, Lcom/android/server/AlarmManagerService;->mAppOps:Landroid/app/AppOpsManager;
- 
-     const/16 v8, 0x2e
- 
-     invoke-virtual/range {p8 .. p8}, Landroid/app/PendingIntent;->getCreatorUid()I
- 
-     move-result v9
- 
-     invoke-virtual/range {p8 .. p8}, Landroid/app/PendingIntent;->getCreatorPackage()Ljava/lang/String;
- 
-     move-result-object v10
- 
-     invoke-virtual {v7, v8, v9, v10}, Landroid/app/AppOpsManager;->checkOpNoThrow(IILjava/lang/String;)I
- 
-     move-result v7
- 
-     if-eqz v7, :cond_6
- 
-     if-nez p1, :cond_9
- 
-     const/16 p1, 0x1
- 
-     :goto_2
-     const/16 v23, 0x1
- 
-     :cond_6
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/AlarmManagerService;->mLock:Ljava/lang/Object;
--- 4086,4091 ----
***************
*** 4203,4209 ****
      move-object/from16 v22, p10
  
      :try_start_0
!     invoke-direct/range {v7 .. v23}, Lcom/android/server/AlarmManagerService;->setImplLocked(IJJJJJLandroid/app/PendingIntent;ZZLandroid/os/WorkSource;Z)V
  
      monitor-exit v28
  
--- 4115,4121 ----
      move-object/from16 v22, p10
  
      :try_start_0
!     invoke-direct/range {v7 .. v22}, Lcom/android/server/AlarmManagerService;->setImplLocked(IJJJJJLandroid/app/PendingIntent;ZZLandroid/os/WorkSource;)V
  
      monitor-exit v28
  
***************
*** 4219,4225 ****
      throw v7
  
      .end local v15    # "maxElapsed":J
-     .end local v23    # "wakeupFiltered":Z
      :cond_7
      const-wide/16 v7, 0x0
  
--- 4131,4136 ----
***************
*** 4242,4253 ****
  
      .restart local v15    # "maxElapsed":J
      goto :goto_1
- 
-     .restart local v23    # "wakeupFiltered":Z
-     :cond_9
-     const/16 p1, 0x3
- 
-     goto :goto_2
  .end method
  
  .method public setTime(J)V
--- 4153,4158 ----
