*** ./smali/com/android/server/BackupManagerService.smali	2015-08-27 23:27:44.175028847 +0800
--- /home/nian/patchrom/m7/temp/new_smali/services.jar.out/./smali/com/android/server/BackupManagerService.smali	2015-08-27 23:29:15.463027631 +0800
***************
*** 35,41 ****
          Lcom/android/server/BackupManagerService$RestoreParams;,
          Lcom/android/server/BackupManagerService$RestoreGetSetsParams;,
          Lcom/android/server/BackupManagerService$ProvisionedObserver;,
!         Lcom/android/server/BackupManagerService$BackupRequest;
      }
  .end annotation
  
--- 35,42 ----
          Lcom/android/server/BackupManagerService$RestoreParams;,
          Lcom/android/server/BackupManagerService$RestoreGetSetsParams;,
          Lcom/android/server/BackupManagerService$ProvisionedObserver;,
!         Lcom/android/server/BackupManagerService$BackupRequest;,
!         Lcom/android/server/BackupManagerService$Injector;
      }
  .end annotation
  
***************
*** 8247,8253 ****
  
      move-object/from16 v0, p0
  
!     invoke-virtual {v0, v15, v2}, Lcom/android/server/BackupManagerService;->startConfirmationUi(ILjava/lang/String;)Z
  
      move-result v2
  
--- 8248,8254 ----
  
      move-object/from16 v0, p0
  
!     invoke-static {v0, v15, v2}, Lcom/android/server/BackupManagerService$Injector;->startConfirmationUi(Lcom/android/server/BackupManagerService;ILjava/lang/String;)Z
  
      move-result v2
  
***************
*** 8333,8339 ****
  
      move-object/from16 v0, p0
  
!     invoke-virtual {v0, v15, v1}, Lcom/android/server/BackupManagerService;->startConfirmationTimeout(ILcom/android/server/BackupManagerService$FullParams;)V
  
      const-string v2, "BackupManagerService"
  
--- 8334,8340 ----
  
      move-object/from16 v0, p0
  
!     invoke-static {v0, v15, v1}, Lcom/android/server/BackupManagerService$Injector;->startConfirmationTimeout(Lcom/android/server/BackupManagerService;ILcom/android/server/BackupManagerService$FullParams;)V
  
      const-string v2, "BackupManagerService"
  
***************
*** 8598,8604 ****
  
      const-string v6, "fullrest"
  
!     invoke-virtual {p0, v5, v6}, Lcom/android/server/BackupManagerService;->startConfirmationUi(ILjava/lang/String;)Z
  
      move-result v6
  
--- 8599,8605 ----
  
      const-string v6, "fullrest"
  
!     invoke-static {p0, v5, v6}, Lcom/android/server/BackupManagerService$Injector;->startConfirmationUi(Lcom/android/server/BackupManagerService;ILjava/lang/String;)Z
  
      move-result v6
  
***************
*** 8678,8684 ****
  
      invoke-virtual {v6, v7, v8, v9}, Landroid/os/PowerManager;->userActivity(JZ)V
  
!     invoke-virtual {p0, v5, v4}, Lcom/android/server/BackupManagerService;->startConfirmationTimeout(ILcom/android/server/BackupManagerService$FullParams;)V
  
      const-string v6, "BackupManagerService"
  
--- 8679,8685 ----
  
      invoke-virtual {v6, v7, v8, v9}, Landroid/os/PowerManager;->userActivity(JZ)V
  
!     invoke-static {p0, v5, v4}, Lcom/android/server/BackupManagerService$Injector;->startConfirmationTimeout(Lcom/android/server/BackupManagerService;ILcom/android/server/BackupManagerService$FullParams;)V
  
      const-string v6, "BackupManagerService"
  
***************
*** 9016,9021 ****
--- 9017,9031 ----
      throw v2
  .end method
  
+ .method getContext()Landroid/content/Context;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/server/BackupManagerService;->mContext:Landroid/content/Context;
+ 
+     return-object v0
+ .end method
+ 
  .method public getCurrentTransport()Ljava/lang/String;
      .locals 3
  
