*** ./smali/com/android/server/BackupManagerService$PerformFullRestoreTask.smali	2015-08-27 23:27:48.479028790 +0800
--- /home/nian/patchrom/m7/temp/new_smali/services.jar.out/./smali/com/android/server/BackupManagerService$PerformFullRestoreTask.smali	2015-08-27 23:29:18.827027587 +0800
***************
*** 1499,1505 ****
      move-result v7
  
      .local v7, "didRead":I
!     if-ltz v7, :cond_0
  
      move-object/from16 v0, p0
  
--- 1499,1505 ----
      move-result v7
  
      .local v7, "didRead":I
!     if-ltz v7, :cond_2
  
      move-object/from16 v0, p0
  
***************
*** 1519,1525 ****
  
      iput-wide v0, v2, Lcom/android/server/BackupManagerService$PerformFullRestoreTask;->mBytes:J
  
-     :cond_0
      const/16 v18, 0x0
  
      move/from16 v0, v18
--- 1519,1524 ----
***************
*** 1860,1920 ****
      move-result-object v11
  
      .local v11, "pkg":Landroid/content/pm/PackageInfo;
-     iget-object v0, v11, Landroid/content/pm/PackageInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
- 
-     move-object/from16 v18, v0
- 
-     move-object/from16 v0, v18
- 
-     iget v0, v0, Landroid/content/pm/ApplicationInfo;->flags:I
- 
-     move/from16 v18, v0
- 
-     const v19, 0x8000
- 
-     and-int v18, v18, v19
- 
-     if-nez v18, :cond_7
- 
-     const-string v18, "BackupManagerService"
- 
-     new-instance v19, Ljava/lang/StringBuilder;
- 
-     invoke-direct/range {v19 .. v19}, Ljava/lang/StringBuilder;-><init>()V
- 
-     const-string v20, "Restore stream contains apk of package "
- 
-     invoke-virtual/range {v19 .. v20}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
- 
-     move-result-object v19
- 
-     move-object/from16 v0, p1
- 
-     iget-object v0, v0, Lcom/android/server/BackupManagerService$FileMetadata;->packageName:Ljava/lang/String;
- 
-     move-object/from16 v20, v0
- 
-     invoke-virtual/range {v19 .. v20}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
- 
-     move-result-object v19
- 
-     const-string v20, " but it disallows backup/restore"
- 
-     invoke-virtual/range {v19 .. v20}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
- 
-     move-result-object v19
- 
-     invoke-virtual/range {v19 .. v19}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
- 
-     move-result-object v19
- 
-     invoke-static/range {v18 .. v19}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
- 
-     const/4 v9, 0x0
- 
-     goto/16 :goto_3
- 
-     :cond_7
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/server/BackupManagerService$PerformFullRestoreTask;->mManifestSignatures:Ljava/util/HashMap;
--- 1859,1864 ----
***************
*** 2550,2568 ****
      iget v6, v0, Landroid/content/pm/ApplicationInfo;->flags:I
  
      .local v6, "flags":I
!     const v18, 0x8000
! 
!     and-int v18, v18, v6
! 
!     if-nez v18, :cond_4
  
!     move-object/from16 v0, p0
! 
!     iget-boolean v0, v0, Lcom/android/server/BackupManagerService$PerformFullRestoreTask;->mNoninteractive:Z
  
!     move/from16 v18, v0
  
!     if-eqz v18, :cond_b
  
      :cond_4
      iget-object v0, v12, Landroid/content/pm/PackageInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
--- 2494,2506 ----
      iget v6, v0, Landroid/content/pm/ApplicationInfo;->flags:I
  
      .local v6, "flags":I
!     #move-object/from16 v0, p0
  
!     #iget-boolean v0, v0, Lcom/android/server/BackupManagerService$PerformFullRestoreTask;->mNoninteractive:Z
  
!     #move/from16 v18, v0
  
!     #if-eqz v18, :cond_b
  
      :cond_4
      iget-object v0, v12, Landroid/content/pm/PackageInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
