*** ./smali/com/android/server/BackupManagerService$PerformFullBackupTask.smali	2015-08-27 23:27:54.983028703 +0800
--- /home/nian/patchrom/m7/temp/new_smali/services.jar.out/./smali/com/android/server/BackupManagerService$PerformFullBackupTask.smali	2015-08-27 23:29:24.247027515 +0800
***************
*** 1492,1497 ****
--- 1492,1499 ----
  
      invoke-virtual {v5}, Ljava/io/FileOutputStream;->close()V
  
+     invoke-static {p2}, Lcom/android/server/BackupManagerService$Injector;->resetManifestFileModifiedTime(Ljava/io/File;)V
+ 
      return-void
  
      .end local v5    # "outstream":Ljava/io/FileOutputStream;
***************
*** 1800,1832 ****
      check-cast v22, Landroid/content/pm/PackageInfo;
  
      .restart local v22    # "pkg":Landroid/content/pm/PackageInfo;
!     move-object/from16 v0, v22
! 
!     iget-object v0, v0, Landroid/content/pm/PackageInfo;->applicationInfo:Landroid/content/pm/ApplicationInfo;
! 
!     move-object/from16 v25, v0
  
!     move-object/from16 v0, v25
  
!     iget v0, v0, Landroid/content/pm/ApplicationInfo;->flags:I
! 
!     move/from16 v25, v0
  
!     const v26, 0x8000
  
-     and-int v25, v25, v26
- 
-     if-nez v25, :cond_3
- 
-     move-object/from16 v0, p0
- 
-     iget-boolean v0, v0, Lcom/android/server/BackupManagerService$PerformFullBackupTask;->mNoninteractive:Z
- 
-     move/from16 v25, v0
- 
-     if-eqz v25, :cond_4
- 
-     :cond_3
      move-object/from16 v0, v22
  
      iget-object v0, v0, Landroid/content/pm/PackageInfo;->packageName:Ljava/lang/String;
--- 1802,1815 ----
      check-cast v22, Landroid/content/pm/PackageInfo;
  
      .restart local v22    # "pkg":Landroid/content/pm/PackageInfo;
!     #move-object/from16 v0, p0
  
!     #iget-boolean v0, v0, Lcom/android/server/BackupManagerService$PerformFullBackupTask;->mNoninteractive:Z
  
!     #move/from16 v25, v0
  
!     #if-eqz v25, :cond_4
  
      move-object/from16 v0, v22
  
      iget-object v0, v0, Landroid/content/pm/PackageInfo;->packageName:Ljava/lang/String;
