*** ./smali/com/android/server/ServerThread.smali	2015-08-27 23:27:53.999028716 +0800
--- /home/nian/patchrom/m7/temp/new_smali/services.jar.out/./smali/com/android/server/ServerThread.smali	2015-08-27 23:29:23.315027527 +0800
***************
*** 788,796 ****
  
      invoke-static {v7, v9}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
!     new-instance v6, Lcom/android/server/LightsService;
  
!     invoke-direct {v6, v5}, Lcom/android/server/LightsService;-><init>(Landroid/content/Context;)V
      :try_end_c
      .catch Ljava/lang/RuntimeException; {:try_start_c .. :try_end_c} :catch_2
  
--- 788,796 ----
  
      invoke-static {v7, v9}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
!     new-instance v6, Lcom/android/server/MiuiLightsService;
  
!     invoke-direct {v6, v5}, Lcom/android/server/MiuiLightsService;-><init>(Landroid/content/Context;)V
      :try_end_c
      .catch Ljava/lang/RuntimeException; {:try_start_c .. :try_end_c} :catch_2
  
***************
*** 941,953 ****
  
      invoke-static {v7, v9}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
!     new-instance v20, Lcom/android/server/input/InputManagerService;
  
      move-object/from16 v0, v20
  
      move-object/from16 v1, v21
  
!     invoke-direct {v0, v5, v1}, Lcom/android/server/input/InputManagerService;-><init>(Landroid/content/Context;Landroid/os/Handler;)V
      :try_end_11
      .catch Ljava/lang/RuntimeException; {:try_start_11 .. :try_end_11} :catch_6a
  
--- 941,953 ----
  
      invoke-static {v7, v9}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
!     new-instance v20, Lcom/android/server/input/MiuiInputManagerService;
  
      move-object/from16 v0, v20
  
      move-object/from16 v1, v21
  
!     invoke-direct {v0, v5, v1}, Lcom/android/server/input/MiuiInputManagerService;-><init>(Landroid/content/Context;Landroid/os/Handler;)V
      :try_end_11
      .catch Ljava/lang/RuntimeException; {:try_start_11 .. :try_end_11} :catch_6a
  
***************
*** 2229,2234 ****
--- 2229,2250 ----
      move-object/from16 v0, v172
  
      invoke-static {v7, v0}, Landroid/os/ServiceManager;->addService(Ljava/lang/String;Landroid/os/IBinder;)V
+ 
+     const-string v7, "miui.usb.service"
+ 
+     new-instance v9, Lcom/miui/server/MiuiUsbService;
+ 
+     invoke-direct {v9, v5}, Lcom/miui/server/MiuiUsbService;-><init>(Landroid/content/Context;)V
+ 
+     invoke-static {v7, v9}, Landroid/os/ServiceManager;->addService(Ljava/lang/String;Landroid/os/IBinder;)V
+ 
+     const-string v7, "miui.os.servicemanager"
+ 
+     new-instance v9, Landroid/os/MiuiServiceManagerInternal;
+ 
+     invoke-direct {v9}, Landroid/os/MiuiServiceManagerInternal;-><init>()V
+ 
+     invoke-static {v7, v9}, Landroid/os/ServiceManager;->addService(Ljava/lang/String;Landroid/os/IBinder;)V
      :try_end_47
      .catch Ljava/lang/Throwable; {:try_start_47 .. :try_end_47} :catch_51
  
***************
*** 2269,2274 ****
--- 2285,2294 ----
      .restart local v155    # "serial":Lcom/android/server/SerialService;
      :cond_19
      :goto_32
+     move/from16 v0, v24
+ 
+     invoke-static {v5, v0}, Lcom/android/server/SystemServerInjector;->addExtraServices(Landroid/content/Context;Z)V
+ 
      :try_start_4a
      const-string v7, "SystemServer"
  
***************
*** 2681,2691 ****
  
      invoke-static {v7, v9}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
!     new-instance v164, Lcom/android/server/ThemeService;
  
!     move-object/from16 v0, v164
  
!     invoke-direct {v0, v5}, Lcom/android/server/ThemeService;-><init>(Landroid/content/Context;)V
      :try_end_5f
      .catch Ljava/lang/Throwable; {:try_start_5f .. :try_end_5f} :catch_37
  
--- 2701,2713 ----
  
      invoke-static {v7, v9}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
!     #new-instance v164, Lcom/android/server/ThemeService;
  
!     #move-object/from16 v0, v164
! 
!     #invoke-direct {v0, v5}, Lcom/android/server/ThemeService;-><init>(Landroid/content/Context;)V
  
!     const/16 v164, 0x0
      :try_end_5f
      .catch Ljava/lang/Throwable; {:try_start_5f .. :try_end_5f} :catch_37
  
***************
*** 2696,2702 ****
  
      move-object/from16 v0, v164
  
!     invoke-static {v7, v0}, Landroid/os/ServiceManager;->addService(Ljava/lang/String;Landroid/os/IBinder;)V
      :try_end_60
      .catch Ljava/lang/Throwable; {:try_start_60 .. :try_end_60} :catch_49
  
--- 2718,2724 ----
  
      move-object/from16 v0, v164
  
!     #invoke-static {v7, v0}, Landroid/os/ServiceManager;->addService(Ljava/lang/String;Landroid/os/IBinder;)V
      :try_end_60
      .catch Ljava/lang/Throwable; {:try_start_60 .. :try_end_60} :catch_49
  
